### PR TITLE
Add statefulness to selectivity of the regions of the Venn diagram, and calculate the percentage of errors out of the total number of instances

### PR DESCRIPTION
**Pull request by ilmarinen**
venn-plain

venn-hover

venn-selected

### CODE REVIEW

**Review by nushib — APPROVED** [review]

**Review by jlema — APPROVED** [review]

File: widget/IntersectionBetweenModelErrors.tsx [hunk 1]
```diff
@@ -238,52 +179,164 @@ class IntersectionBetweenModelErrors extends Component<IntersectionBetweenModelE
...
-          .attr('transform',
-              "translate(" +
-              xCenter2 + "," +
-              yCenter + ")")
-          .attr("fill", "rgba(206, 160, 205, 0.8)")
+        var r = Math.min(Ra, Rb);
+        var R = Math.max(Ra, Rb);
+
+        // We perform the bisection search between the two extreme values
+        let d = bisect(aIntersection, (r + R - 0.00001), (R - r + 0.00001));
```
Comment (Nking92) [inline widget/IntersectionBetweenModelErrors.tsx:249]: Sometimes bisect fails for me with the error "f(a) and f(b) must evaluate to opposite signs". Is that never supposed to happen? Is there a way to make the code more robust so the widget doesn't crash if bisect throws an error?

**Review by Nking92 — APPROVED** [review]
Looks great! I really appreciate the comments you've added. The math was hard to follow without some description of the geometry. The overall code quality is much better as well. It's easier to understand what's going on now.

File: widget/IntersectionBetweenModelErrors.tsx (hunk 1 above)
Comment (ilmarinen) [reply widget/IntersectionBetweenModelErrors.tsx:249]: Good question. I noticed this too, and am working on a fix.

Basically the case when this happens is likely one of the two extremal cases, i.e. empty intersection or an intersection that has one set wholly included in the other.

It should be possible to fix this by checking ahead of time and not running the bisect procedure when we are in either of those regimes.
